src/renderer/entities/wallet/lib/account-utils.ts

```diff
@@ -83,3 +78,25 @@ function getAllAccountIds(accounts: Account[], chainId: ChainId): AccountId[] {
 function getWalletAccounts<T extends Account>(walletId: Wallet['id'], accounts: T[]): T[] {
   return accounts.filter((account) => account.walletId === walletId);
 }
+
+function getAccountsWithGroupedShards<T extends Account>(accounts: T[]): Array<T | Array<T>> {
+  const shardsIndexes: Record<ShardAccount['group'], number> = {};
```

Comment:
maybe use `Set` here?

in this case, we can check the following conditions with: `shardsSet.has(account.group)`

Reply:
Good solution `set.has` has `O(1)` time complexity, but I think here we store indexes for shards array, and can't use such solution

Reply:
yeah, we need pairs of groupId - index so we cant use set here